code/modules/antagonists/heretic/magic/moon_parade.dm

```diff
@@ -0,0 +1,100 @@
+/datum/action/cooldown/spell/pointed/projectile/moon_parade
+	name = "Lunar parade"
+	desc = "This unleashes the parade towards a target."
+	background_icon_state = "bg_heretic"
+	overlay_icon_state = "bg_heretic_border"
+	button_icon = 'icons/mob/actions/actions_ecult.dmi'
+	button_icon_state = "moon_parade"
+	ranged_mousepointer = 'icons/effects/mouse_pointers/moon_target.dmi'
+
+	sound = 'sound/magic/cosmic_energy.ogg'
+	school = SCHOOL_FORBIDDEN
+	cooldown_time = 30 SECONDS
+
+	invocation = "L'N'R P'RAD"
+	invocation_type = INVOCATION_SHOUT
+	spell_requirements = NONE
+
+	active_msg = "You prepare to make them join the parade!"
+	deactive_msg = "You stop the music and halt the parade... for now."
+	cast_range = 12
+	projectile_type = /obj/projectile/moon_parade
+
+
+/obj/projectile/moon_parade
+	name = "Lunar parade"
+	icon_state = "lunar_parade"
+	damage = 0
+	damage_type = BURN
+	speed = 1
+	range = 75
+	ricochets_max = 40
+	ricochet_chance = 500
+	ricochet_incidence_leeway = 0
+	pixel_speed_multiplier = 0.2
+	projectile_piercing = PASSMOB|PASSVEHICLE
+	///looping sound datum for our projectile.
+	var/datum/looping_sound/moon_parade/soundloop
+	// A list of the people we hit
+	var/list/mobs_hit = list()
```

Comment:
Mobs tracked in this list will hard delete, you need to either use weakrefs or register a qdel signal. 

IE: 
```dm
mobs_hit |= WEAKREF(victim)
``` 
```dm
for(var/datum/weakref/mob_ref in mobs_hit) 
    var/mob/real_mob = mob_ref.resolve()
```

Or

```dm
RegisterSignal(victim, COMSIG_QDELETED, PROC_REF(clear_reference)
```
```dm
/obj/projectile/moon_parade/proc/clear_reference(datum/source)
    SIGNAL_HANDLER
    mobs_hit -= source
```

Reply:
Went with weakref